validator/sawtooth_validator/gossip/permission_verifier.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+import logging
+# pylint: disable=import-error,no-name-in-module
+# needed for google.protobuf import
+from google.protobuf.message import DecodeError
+
+from sawtooth_validator.protobuf import client_pb2
+from sawtooth_validator.protobuf.batch_pb2 import BatchHeader
+from sawtooth_validator.protobuf.validator_pb2 import Message
+from sawtooth_validator.networking.dispatch import HandlerResult
+from sawtooth_validator.networking.dispatch import HandlerStatus
+from sawtooth_validator.networking.dispatch import Handler
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+def check_batches_signing_keys(batches, allowed_pubkeys):
+    for batch in batches:
+        if not check_batch_signing_key(batch, allowed_pubkeys):
+            return False
+    return True
```

Comment:
As mentioned in a previous review, this could be shortened to:
`return all(check_batch_signing_key(batch, allowed_pubkeys) for batch in batches)`